python/tests/test_datasetslice.py

```diff
@@ -11,276 +11,363 @@
 
 class TestDatasetSlice(unittest.TestCase):
     def setUp(self):
-        d = sc.Dataset()
-        d[sc.Coord.X] = (['x'], np.arange(10))
-        d[sc.Data.Value, "a"] = (['x'], np.arange(10))
-        d[sc.Data.Value, "b"] = (['x'], np.arange(10))
+        var = sc.Variable(['x'], values=np.arange(10))
+        d = sc.Dataset(data={'a': var, 'b': var}, coords={'x': var})
         self._d = d
 
+    # Leave as is
     def test_type(self):
-        ds_slice = self._d.subset["a"]
-        self.assertEqual(type(ds_slice), sc.DatasetSlice)
-
-    def test_extract_slice(self):
-        ds_slice = self._d.subset["a"]
-        self.assertEqual(type(ds_slice), sc.DatasetSlice)
-        # We should have just one data variable
-        self.assertEqual(
-            1, len([var for name, tag, var in ds_slice if tag.is_data]))
-        # We should have just one coord variable
-        self.assertEqual(
-            1, len([var for name, tag, var in ds_slice if tag.is_coord]))
-        self.assertEqual(2, len(ds_slice))
-
-    def test_slice_back_ommit_range(self):
-        sl = self._d['x', 1:-1][sc.Data.Value, "a"].numpy
+        ds_slice = self._d['a']
+        self.assertEqual(type(ds_slice), sc.DataArrayView)
+
+    def test_slice_with_range_datasetview_then_dataarrayview(self):
+        sl = self._d['x', 1:-1]['a'].values
         ref = np.array([1, 2, 3, 4, 5, 6, 7, 8], dtype=np.int64)
         self.assertEqual(ref.shape, sl.shape)
-        self.assertEqual(np.allclose(sl, ref), True)
+        self.assertTrue(np.allclose(sl, ref))
```

Comment:
Not sure I understand the reasoning behind the testing here and below: Why are subcomponents such as shape and values compared directly? Is there anything wrong with using `__eq__`, i.e., just
```python
assert self._d['x', 1:-1]['a'] == ref
```
?